packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -96,13 +96,13 @@ exports.onPostBuild = (args, pluginOptions) => {
     runtimeCaching: [
       {
         // Add runtime caching of various page resources.
-        urlPattern: /\.(?:png|jpg|jpeg|webp|svg|gif|tiff|js|woff|woff2|json|css)$/,
+        urlPattern: /^https?:.*\.(?:png|jpg|jpeg|webp|svg|gif|tiff|js|woff|woff2|json|css)$/,
         handler: `staleWhileRevalidate`,
       },
       {
-        // Use the Network First handler for external resources
-        urlPattern: /^https?:/,
-        handler: `networkFirst`,
+        // Google Fonts CSS (doesn't end in .css so we need to specify it)
+        urlPattern: /^https?:\/\/fonts\.googleapis\.com\/css/,
```

Comment:
Maybe just use a regex like in this recipe? https://gist.github.com/addyosmani/0e1cfeeccad94edc2f0985a15adefe54#cache-all-google-fonts-requests-up-to-a-maximum-of-30-cache-entries

Also from what I understand, staleWhileRevalidate will request both cache/network each time, and then update the cache on success.

It seems reasonable that something like google fonts can hit the cache first, then the network if that fails? What do you think?

Reply:
`staleWhileRevalidate` in the Workbox config is equivalent to the `CacheFirst` strategy when calling Workbox APIs directly, so the current config actually does what you've described (except for the max 30 entries). `woff2` and other font files from Google (or any domain) are covered by the first regex, which matches common extensions (and now works for external resources due to the added `^https?:.*` at the start).

Also, since the linked code calls Workbox APIs directly, we'd have to move it to `sw-append.js`, which means it can't be customised by the user - so I think it's best to leave it user configurable. Basically Workbox generates similar code by reading the config - `sw-append.js` is for extra stuff which we can't do with the config.